src/components/InstructionsErrorBoundary.jsx

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import markdown from '../util/markdown';
+
+export default class InstructionsErrorBoundary extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {useHighlighting: true};
+  }
+
+  unstable_handleError() { // eslint-disable-line camelcase
+    /*
+    Highlight.js throws an exception when fed a language it doesn't have
+    configured. None of the React wrappers we're using catch that error,
+    and since it gets run in the child React components created from markdown
+    code sections, these exceptions occur outside the current callstack.
+    `unstable_handleError` is a kludge for catching exceptions raised in child
+    renders introduced in React 15 and to be replaced by `componentDidCatch` in
+    React 16.
+    <InstructionsErrorBoundary> exists because `unstable_handleError` only
+    wraps the *initial* mounting of a child -- we need an "error boundary"
+    component that comes into being in the same mounting phase that
+    Highlight.js gets run.
+    In theory this component can be removed and replaced with a
+    `componentDidCatch` method on the parent when we upgrade to React 16.
+     */
```

Comment:
I’m generally much more into function/method/class-level comments than inline ones… in this case just a matter of moving the comment block a couple lines north, you down?

Reply:
I'm assuming you mean put it immediately above `unstable_handleError`, instead of immediately below? To be honest I interpret both as "method-level" but it's no problem to move it.

Reply:
Yep, I guess just from the way jsdoc and equivalent systems do it, putting it above the method header feels a bit more natural